src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+      addImport(compUnit, typeName);
+      typeNameMap.put(typeName, getSimpleTypeName(typeName));
+    }
+
+    CompilationUnit result = compUnit;
+    List<ImportDeclaration> imports = result.getImports();
+    for (String type : typeNameMap.keySet()) {
+      String compUnitStr = result.toString();
+
+      // Replace all instances of the fully-qualified type name with the simple type name.
+      if (compUnitStr.contains(type)) {
+        compUnitStr = compUnitStr.replace(type, typeNameMap.get(type));
+      }
+      CompilationUnit compUnitWithSimpleTypeNames;
+      try {
+        // Parse the compilation unit and set the imports.
+        compUnitWithSimpleTypeNames = JavaParser.parse(IOUtils.toInputStream(compUnitStr, "UTF-8"));
+        compUnitWithSimpleTypeNames.setImports(imports);
+      } catch (ParseException e) {
+        System.err.println("Error parsing into a compilation unit.");
+        continue;
+      } catch (IOException e) {
+        System.err.println("IOUtils error creating input stream from string.");
+        continue;
+      }
+      // Check that the simplification is correct.
+      String newFilePath = writeToFile(result, filePath, "Minimized");
+      if (checkCorrectlyMinimized(
+          newFilePath, classpath, packageName, expectedOutput, timeoutLimit)) {
+        result = compUnitWithSimpleTypeNames;
+      }
+    }
+    return result;
+  }
+
+  /**
+   * Get the simple type name of a fully qualified type name. For example, {@code java.lang.String}
+   * should be simplified to {@code String}.
+   *
+   * @param typeName fully qualified type name
+   * @return simple type name of {@code typeName}
+   */
+  public static String getSimpleTypeName(String typeName) {
+    int indexOflastSeparator = typeName.lastIndexOf('.');
+    return indexOflastSeparator > 0
+        ? typeName.substring(indexOflastSeparator + 1, typeName.length())
+        : typeName;
+  }
+
+  /**
+   * Check if a Java file has been correctly minimized. The file should not have compilation errors
+   * or runtime errors. The file should fail in the same way as the original file.
+   *
+   * @param filePath the absolute path to the Java file
+   * @param classpath classpath needed to compile/run Java file
+   * @param packageName the name of the package that the Java file is in
+   * @param expectedOutput expected output of running JUnit test suite
+   * @param timeoutLimit number of seconds allowed for the whole test suite to be run
+   * @return true if there are no compilation and no runtime errors and the output is equal to the
+   *     expected output
+   */
+  private static boolean checkCorrectlyMinimized(
+      String filePath,
+      String classpath,
+      String packageName,
+      String expectedOutput,
+      int timeoutLimit) {
+    Results res = compileAndRun(filePath, classpath, packageName, timeoutLimit);
+
+    return res.compOut.exitValue == 0
+        && res.runOut != null
+        && res.runOut.errout.isEmpty()
```

Comment:
I tried the above approach, however there were various issues that arose. Running the JUnit test suite programmatically required the minimizer to pass in the `Class` object for the test suite. I attempted to use Class.forName(...) and ClassLoader.loadClass(...), but these both resulted in `ClassNotFoundExceptions`. After some searching, it looked like the issue was with the ClassLoader and not being able to dynamically load the new classes that were being generated by the minimizer. When I ran the minimizer the first time, an exception would occur (after the new file was written and compiled). When I then ran it a second time, there wouldn't be an exception since the file existed. However, the minimizer writes to this file many times, but the one that gets used is apparently the cached copy of the original. The new approach is to use the existing approach of running the JUnit test suite, but parsing the standard output to produce a map from method name to failure stack trace. This time I made sure to handle line numbers.